exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -285,16 +325,7 @@ func (prwe *prwExporter) export(ctx context.Context, requests []*prompb.WriteReq
 	return errs
 }
 
-func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequest) error {
-	buf := bufferPool.Get().(*buffer)
-	buf.protobuf.Reset()
-	defer bufferPool.Put(buf)
-
-	// Uses proto.Marshal to convert the WriteRequest into bytes array
-	errMarshal := buf.protobuf.Marshal(writeReq)
-	if errMarshal != nil {
-		return consumererror.NewPermanent(errMarshal)
-	}
+func (prwe *prwExporter) execute(ctx context.Context, buf *buffer) error {
```

Comment:
I didn't understand this change. Could you clarify why we are getting the buffer outside this?

This is being duplicated in several places now 🤔

Reply:
the benefit is we can use a single `execute` function because both v1 and v2 pass it a buffer instead of passing it the request struct which is different between v1 and v2.

Reply:
yeah so even moving it to a separate function won't work since you need to pass it the different request formats.